File: Exercicios_diarios/Revisão_intermediaria/Subsequência.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int i = 0;

        if (s.size() > t.size())
            return false;
        if (s.size() == 0)
            return true;

        for (int j = 0; j < t.size(); j++)
            if (i <= s.size() - 1)
                if (s[i] == t[j])
                    i ++;

        return i == s.size();
    }
};

File: Python/Exercicios_diarios/Revisão_intermediaria/Subsequência.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0

        if len(s) > len(t):
            return False
        if len(s) == 0:
            return True
                
        for j in range(0, len(t)):
            if i <= len(s) - 1:
                if s[i] == t[j]:
                    i += 1
                    
        return i == len(s)
    
